src/Microsoft.DocAsCode.Build.Engine/HostService.cs

```diff
@@ -81,11 +81,43 @@ public ImmutableList<FileModel> LookupByUid(string uid)
             }
         }
 
+        public MarkupResult MarkupMultiple(string markdown, FileAndType ft)
+        {
+            try
+            {
+                var html = MarkdownService.Markup(markdown, ft.File);
+                var result = new MarkupResult
+                {
+                    Html = html
+                };
+
+                var parts = YamlHtmlPart.SplitYamlHtml(html);
+                var docs = parts.Select(part => part.Doc);
+                var mrs = docs.Select(doc => MarkupCore(doc, ft));
+                foreach (var mr in mrs)
+                {
+                    result.LinkToFiles = result.LinkToFiles.Union(mr.LinkToFiles).ToImmutableArray();
+                    result.LinkToUids = result.LinkToUids.Union(mr.LinkToUids);
+                }
+                return result;
```

Comment:
yes, what this method really do is to transform markdown to html and collect links.

BTW, this method will be refactored in the next iteration according to vwxyzh's advice

---
In reply to: [65125046](https://github.com/dotnet/docfx/pull/343#discussion_r65125046) [](ancestors = 65125046)